From 2eda75ed21dcf929327c65e8e21ad8afc2e8c998 Mon Sep 17 00:00:00 2001
From: Zhipeng Gong <zhipeng.gong@intel.com>
Date: Tue, 1 Mar 2016 08:26:46 -0500
Subject: [PATCH 093/153] [VPG]: drm/i915: export edram cap register to user space

Starting in SKL, eDRAM was moved to the memory side of the memory control. So, it is
not considered a "architectural" cache, which is usually reported in cpuid. So, we have
to read eDRAM info from a MMIO register now.
---
 drivers/gpu/drm/i915/intel_uncore.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_uncore.c b/drivers/gpu/drm/i915/intel_uncore.c
index 43cba12..d62bfbd 100644
--- a/drivers/gpu/drm/i915/intel_uncore.c
+++ b/drivers/gpu/drm/i915/intel_uncore.c
@@ -1268,6 +1268,7 @@ static const struct register_whitelist {
 	uint32_t gen_bitmask;
 } whitelist[] = {
 	{ RING_TIMESTAMP(RENDER_RING_BASE), 8, GEN_RANGE(4, 9) },
+	{ HSW_EDRAM_PRESENT, 4, GEN_RANGE(7, 9) },
 };
 
 int i915_reg_read_ioctl(struct drm_device *dev,
-- 
1.7.1

